src/node_file.cc

```diff
@@ -1341,8 +1352,24 @@ int MKDirpAsync(uv_loop_t* loop,
           if (err == UV_EEXIST &&
               req_wrap->continuation_data->paths.size() > 0) {
             uv_fs_req_cleanup(req);
-            MKDirpAsync(loop, req, path.c_str(),
-                        req_wrap->continuation_data->mode, nullptr);
+            int err = uv_fs_stat(loop, req, path.c_str(),
```

Comment:
This logic is getting chunky enough, I wonder if we could break it out into a helper method.

Reply:
Refactored a bit to make the code more compact. PTAL.